redbot/cogs/mutes/mutes.py

```diff
@@ -0,0 +1,1219 @@
+                to_clear.append(c_id)
+            await self.config.channel(channel).muted_users.set(self._channel_mutes[c_id])
+        for c_id in to_clear:
+            await self.config.channel_from_id(c_id).muted_users.clear()
+
+    async def _auto_channel_unmute_user(self, channel: discord.TextChannel, data: dict):
+        delay = data["until"] - datetime.now(timezone.utc).timestamp()
+        if delay < 1:
+            delay = 0
+        await asyncio.sleep(delay)
+        try:
+            member = channel.guild.get_member(data["member"])
+            author = channel.guild.get_member(data["author"])
+            if not member or not author:
+                return
+            success = await self.channel_unmute_user(
+                channel.guild, channel, author, member, _("Automatic unmute")
+            )
+            if success["success"]:
+                try:
+                    await modlog.create_case(
+                        self.bot,
+                        channel.guild,
+                        datetime.now(timezone.utc),
+                        "cunmute",
+                        member,
+                        author,
+                        _("Automatic unmute"),
+                        until=None,
+                    )
+                except RuntimeError as e:
+                    log.error(_("Error creating modlog case"), exc_info=e)
+        except discord.errors.Forbidden:
+            return
+
+    @commands.Cog.listener()
+    async def on_member_update(self, before: discord.Member, after: discord.Member):
+        """
+        Used to handle the cache if a member manually has the muted role removed
+        """
+        guild = before.guild
+        if guild.id not in self.mute_role_cache:
+            return
+        mute_role_id = self.mute_role_cache[guild.id]
+        mute_role = guild.get_role(mute_role_id)
+        if not mute_role:
+            return
+        b = set(before.roles)
+        a = set(after.roles)
+        roles_removed = list(b - a)
+        roles_added = list(a - b)
+        if mute_role in roles_removed:
+            # send modlog case for unmute and remove from cache
+            if guild.id not in self._server_mutes:
+                # they weren't a tracked mute so we can return early
+                return
+            if after.id in self._server_mutes[guild.id]:
+                try:
+                    await modlog.create_case(
+                        self.bot,
+                        guild,
+                        datetime.utcnow(),
+                        "sunmute",
+                        after,
+                        None,
+                        _("Manually removed mute role"),
+                    )
+                except RuntimeError as e:
+                    log.error(_("Error creating modlog case"), exc_info=e)
+                del self._server_mutes[guild.id][after.id]
+        if mute_role in roles_added:
```

Comment:
Role can't be both removed and added, so we can make this into an `elif`:
```suggestion
        elif mute_role in roles_added:
```